includes/class-wc-stripe-webhook-handler.php

```
@@ -1827,6 +1844,128 @@ protected function handle_deferred_payment_intent_succeeded( $order, $intent_id
+
+		if ( WC_Stripe_Intent_Status::CANCELED === $intent->status ) {
+			WC_Stripe_Logger::debug( "Skipped refunding cancelled order {$order->get_id()} for Stripe PaymentIntent {$intent_id} - the intent is already cancelled." );
+			return;
+		}
+
+		$charge    = $this->get_latest_charge_from_intent( $intent );
+		$charge_id = isset( $charge->id ) ? $charge->id : '';
+
+		if ( $charge && ! empty( $charge->refunded ) ) {
+			$this->mark_cancelled_order_refunded( $order );
+			WC_Stripe_Logger::debug( "Skipped refunding cancelled order {$order->get_id()} for Stripe PaymentIntent {$intent_id} - the charge is already refunded." );
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'WC_Stripe_Logger' in method 'handle_deferred_payment_for_cancelled_order'. (undefined)

(StaticAccess)